processor/src/decorators/mod.rs

```diff
@@ -186,6 +188,65 @@ impl Process {
 
         Ok(())
     }
+
+    /// Given evaluations of a polynomial over some specified domain, this routine
+    /// interpolates the evaluations into a polynomial in coefficient form and writes
+    /// the result into the advice tape. The interpolation is performed using the iNTT
+    /// algorithm. The evaluations are expected to be in the quadratic extension field of
+    /// Z_q and the resulting coefficients are in the quadratic extension field as
+    /// well | q = 2^64 - 2^32 + 1.
+    ///
+    /// Input stack state should look like
+    ///
+    /// `[output_poly_len, input_eval_len, input_eval_begin_address, ...]`
+    ///
+    /// - `output_poly_len` must be a power of 2.
+    /// - `input_eval_len` must be a power of 2.
+    /// - `output_poly_len <= input_eval_len`
+    /// - Only starting memory address of evaluations ( of length `input_eval_len` ) is
+    /// provided on the stack, consecutive memory addresses are expected to be holding
+    /// remaining `input_eval_len - 2` many evaluations.
+    /// - Each memory address holds two evaluations of the polynomial at adjacent points
+    ///
+    /// Final advice tape should look like
+    ///
+    /// `[coeff_0, coeff_1, ..., coeff_{n-1}, ...]` | n = output_poly_len
+    ///
+    /// Program which is requesting this non-deterministic computation should read `coeff0`
+    /// first i.e. `coeff{n-1}` should be seen at the very end.
+    fn inject_ext2_intt_result(&mut self) -> Result<(), ExecutionError> {
+        let out_poly_len = self.stack.get(0).as_int() as usize;
+        let in_evaluations_len = self.stack.get(1).as_int() as usize;
+        let in_evaluations_addr = self.stack.get(2).as_int();
+
+        if !in_evaluations_len.is_power_of_two() {
+            return Err(ExecutionError::NttDomainSizeNotPowerof2(in_evaluations_len as u64));
+        }
+        if !out_poly_len.is_power_of_two() {
+            return Err(ExecutionError::NttDomainSizeNotPowerof2(out_poly_len as u64));
+        }
```

Comment:
Sorry, I missed this in the previous review: we probably don't need to check that output length is a power of two - it could be something else as well.

But, I think we might want to check:
- Output length is smaller than or equal to the input length.
- Input length is greater than 1.

Reply:
I've introduced another error type ( with some metadata ) for doing this, please check 3f9de5858fb847aa6e866a3eac121f05e4ddef68.

Reply:
I would probably have 3 different error types here:
- `NttDomainSizeTooSmall` - when domain size is 0 or 1.
- `NttDomainSizeNotPowerOf2` - when input size is not a power of two (but bigger than 1).
- `InterpolationResultSizeTooBig` - when result size is bigger than input size.

Reply:
Thanks, more self-explanatory now. Fixed in afea5111